scripts/create-tenant.sh

```diff
@@ -0,0 +1,231 @@
+
+Requirements:
+
+ * You must login with the cf client with an "admin" user.
+ * You must have a functional aws client with credentials configured.
+
+Where:
+  -r           Delete/recreate the user. The user will be recreated
+               and the password reset.
+
+  -e <email>   User email to add and configure as organization and
+               space manager. If the user is created or recreated,
+               they will receive a mail with the new password.
+
+  -o <orgname> Organisation to create and add the user to. If the
+               organisation already exists the script will carry on.
+
+EOF
+  exit 1
+}
+
+load_colors() {
+  if [ -t 1 ] ; then
+    export ESC_SEQ="\x1b["
+    export COL_RESET="${ESC_SEQ}39;49;00m"
+    export COL_LIGHT="${ESC_SEQ}1m"
+    export COL_DIM="${ESC_SEQ}2m"
+    export COL_BLINK="${ESC_SEQ}5m"
+    export COL_RED="${ESC_SEQ}31m"
+    export COL_GREEN="${ESC_SEQ}32m"
+    export COL_YELLOW="${ESC_SEQ}33m"
+    export COL_BLUE="${ESC_SEQ}34m"
+    export COL_MAGENTA="${ESC_SEQ}35m"
+    export COL_CYAN="${ESC_SEQ}36m"
+  fi
+}
+
+abort() {
+  echo -e "${COL_RED:-}${COL_LIGHT:-}ERROR:${COL_RESET:-} $*" 1>&2
+  exit 1
+}
+
+abort_usage() {
+  echo -e "${COL_RED:-}${COL_LIGHT:-}ERROR:${COL_RESET:-} $*" 1>&2
+  usage
+  exit 1
+}
+
+
+check_params_and_environment() {
+  if [ -z "${EMAIL:-}" ]; then
+    abort_usage "Email must be defined"
+  fi
+
+  local email_expr="^[A-Za-z0-9._%+-]+@[A-Za-z0-9.-]+\.[A-Za-z]{2,4}$"
+  if ! [[ "${EMAIL}" =~ ${email_expr} ]]; then
+    abort "You must specify a valid email"
+  fi
+
+  if [ -z "${ORG:-}" ]; then
+    abort_usage "Org must be defined"
+  fi
+
+  if ! aws ses get-send-quota >/dev/null 2>&1; then
+    abort "You must have AWS cli installed and configured with valid credentials. Test it with: aws ses get-send-quota"
+  fi
+
+}
+
+generate_password() {
+  PASSWORD=$(tr -dc _A-Z-a-z-0-9 < /dev/urandom | head -c32 || true)
+  if [[ -z "${PASSWORD}" ]]; then
+    abort "Failure generating password"
+  fi
+}
+
+create_org_space() {
+  cf create-org "${ORG}"
+  cf create-space "${DEFAULT_SPACE}" -o "${ORG}"
+}
+
+create_user() {
+  if [[ "${RESET_USER}" == "true" ]]; then
+    cf delete-user "${EMAIL}" -f
+  fi
+
+  if cf create-user "${EMAIL}" "${PASSWORD}" | tee "${TMP_OUTPUT}"; then
+    if ! grep -q "already exists" "${TMP_OUTPUT}"; then
+      SEND_EMAIL=true
+    fi
+  else
+    abort "Error creating user ${EMAIL}"
+  fi
+}
+
+set_user_roles() {
+  cf set-org-role "${EMAIL}" "${ORG}" OrgManager
+  cf set-space-role "${EMAIL}" "${ORG}" "${DEFAULT_SPACE}" SpaceDeveloper
+  cf set-space-role "${EMAIL}" "${ORG}" "${DEFAULT_SPACE}" SpaceManager
+}
+
+# Expand variables from subject, escaping quotes
+get_subject() {
+  eval "echo ${SUBJECT}" | \
+    sed 's/\([^\/]\)"/\1\\"/g'
+}
+
+# Expand variables from message body, escaping new lines and quotes
+get_body() {
+  eval "echo \"${MESSAGE}\"" | \
+    sed ':a;N;$!ba;s/\n/\\n/g'| \
+    sed 's/\([^\/]\)"/\1\\"/g'
```

Comment:
These don't do the right thing with BSD sed (as found on OS X).

The email sending results in an error:
```
Error parsing parameter '--message': Invalid JSON: Invalid control character at: line 9 column 26 (char 170)
JSON received:
    {
      "Subject": {
        "Data": "Welcome to the Government PaaS",
        "Charset": "utf8"
      },
      "Body": {
        "Text": {
          "Data": "Hello,

Your account has been created for the  Government PaaS service.

Your organisation is \"alext-test\" and your login and password:

 - login: me@example.com
 - password: super_secret

To begin using the PaaS you will need to install the cf command line client
and execute the following commands.

cf target https://api.cloud.service.gov.uk
cf login

Further documentation on the cf command line client is available from
http://docs.cloudfoundry.org/cf-cli/install-go-cli.html


Regards,
Government PaaS team.",
          "Charset": "utf8"
        }
      }
    }
```